app/src/main/kotlin/com/backbase/golden_sample_app/user/ProfileRepository.kt

```diff
@@ -0,0 +1,49 @@
+package com.backbase.golden_sample_app.user
+
+import com.backbase.android.client.usermanagerclient2.api.UserProfileManagementApi
+import com.backbase.android.client.usermanagerclient2.model.UserProfile
+import com.backbase.android.clients.common.CallResult
+import com.backbase.android.clients.common.coroutines.executeAsSuspended
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+
+class ProfileRepository(
+    private val userProfileManagementApi: UserProfileManagementApi,
+    private val inMemoryDataSource: InMemoryProfileDataSource = InMemoryProfileDataSource(),
```

Comment:
What is the value of wrapping the UserProfile in such an inMemory class?

Reply:
the `getOwnUserProfile` function receives a `forceUpdate` parameter. If it is false, it will return the value from the memory or null if the cache is empty.